id: PYSEC-0000-CVE-2022-23583
details: Tensorflow is an Open Source Machine Learning Framework. A malicious user
  can cause a denial of service by altering a `SavedModel` such that any binary op
  would trigger `CHECK` failures. This occurs when the protobuf part corresponding
  to the tensor arguments is modified such that the `dtype` no longer matches the
  `dtype` expected by the op. In that case, calling the templated binary operator
  for the binary op would receive corrupted data, due to the type confusion involved.
  If `Tin` and `Tout` don't match the type of data in `out` and `input_*` tensors
  then `flat<*>` would interpret it wrongly. In most cases, this would be a silent
  failure, but we have noticed scenarios where this results in a `CHECK` crash, hence
  a denial of service. The fix will be included in TensorFlow 2.8.0. We will also
  cherrypick this commit on TensorFlow 2.7.1, TensorFlow 2.6.3, and TensorFlow 2.5.3,
  as these are also affected and still in supported range.
affected:
- package:
    name: tensorflow-cpu
    ecosystem: PyPI
    purl: pkg:pypi/tensorflow-cpu
  ranges:
  - type: GIT
    repo: https://github.com/tensorflow/tensorflow
    events:
    - introduced: "0"
    - fixed: a7c02f1a9bbc35473969618a09ee5f9f5d3e52d9
  - type: ECOSYSTEM
    events:
    - introduced: "0"
    - fixed: 2.5.3
    - introduced: 2.6.0
    - fixed: 2.6.3
references:
- type: WEB
  url: https://github.com/tensorflow/tensorflow/blob/a1320ec1eac186da1d03f033109191f715b2b130/tensorflow/core/kernels/cwise_ops_common.h#L88-L137
- type: ADVISORY
  url: https://github.com/tensorflow/tensorflow/security/advisories/GHSA-gjqc-q9g6-q2j3
- type: FIX
  url: https://github.com/tensorflow/tensorflow/commit/a7c02f1a9bbc35473969618a09ee5f9f5d3e52d9
aliases:
- CVE-2022-23583
- GHSA-gjqc-q9g6-q2j3
modified: "2022-02-10T17:15:00Z"
published: "2022-02-04T23:15:00Z"
